File: Flutter - Non-Native/discover_destination/lib/repository/users_repository.dart
import 'package:discover_destination/model/user.dart';
import 'package:discover_destination/service/users_service.dart';

class UsersRepository{
  UsersService service;

  UsersRepository(this.service);

  bool checkUser(User user) {
    return service.checkUser(user);
  }
}

File: Android - Client-Server/client/discover_destination/app/src/main/java/com/example/discover_destination/repository/UsersRepository.kt
package com.example.discover_destination.repository

import com.example.discover_destination.databases.UsersDao
import com.example.discover_destination.domain.User
import com.example.discover_destination.service.UsersService
import javax.inject.Inject

class UsersRepository @Inject constructor(
    val service: UsersDao
) {
    fun checkUser(user: User): Boolean {
        return service.checkUser(user.username, user.password)
    }
}